addons/binding/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/handler/HeosPlayerHandler.java

```diff
@@ -0,0 +1,155 @@
+        executerPool.schedule(new InitializationRunnable(), 3, TimeUnit.SECONDS);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public PercentType getNotificationSoundVolume() {
+        return PercentType.valueOf(player.getLevel());
+    }
+
+    @Override
+    public void setNotificationSoundVolume(PercentType volume) {
+        api.setVolume(volume.toString(), pid);
+    }
+
+    @Override
+    public void playerStateChangeEvent(String pid, String event, String command) {
+        if (pid.equals(this.pid)) {
+            handleThingStateUpdate(event, command);
+        }
+    }
+
+    @Override
+    public void playerMediaChangeEvent(String pid, Map<String, String> info) {
+        this.player.updateMediaInfo(info);
+        if (pid.equals(this.pid)) {
+            handleThingMediaUpdate(info);
+        }
+    }
+
+    @Override
+    public void bridgeChangeEvent(String event, String result, String command) {
+        // Do nothing
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void setStatusOffline() {
+        api.unregisterforChangeEvents(this);
+        updateStatus(ThingStatus.OFFLINE);
+    }
+
+    @Override
+    protected void updateHeosThingState() {
+        player = heos.getPlayerState(pid);
+    }
+
+    @Override
+    protected void refreshChannels() {
+        if (player.getLevel() != null) {
+            updateState(CH_ID_VOLUME, PercentType.valueOf(player.getLevel()));
+        }
+
+        if (player.getMute().equals(ON)) {
+            updateState(CH_ID_MUTE, OnOffType.ON);
+        } else {
+            updateState(CH_ID_MUTE, OnOffType.OFF);
+        }
+
+        if (player.getState().equals(PLAY)) {
+            updateState(CH_ID_CONTROL, PlayPauseType.PLAY);
+        }
+        if (player.getState().equals(PAUSE) || player.getState().equals(STOP)) {
+            updateState(CH_ID_CONTROL, PlayPauseType.PAUSE);
+        }
+        if (player.getShuffle().equals(HeosConstants.HEOS_ON)) {
+            updateState(CH_ID_SHUFFLE_MODE, OnOffType.ON);
+        }
+        if (player.getShuffle().equals(HeosConstants.HEOS_OFF)) {
+            updateState(CH_ID_SHUFFLE_MODE, OnOffType.OFF);
+        }
+        updateState(CH_ID_SONG, StringType.valueOf(player.getSong()));
+        updateState(CH_ID_ARTIST, StringType.valueOf(player.getArtist()));
+        updateState(CH_ID_ALBUM, StringType.valueOf(player.getAlbum()));
+        updateState(CH_ID_COVER, StringType.valueOf(player.getImageUrl()));
+        updateState(CH_ID_STATION, StringType.valueOf(player.getStation()));
+        updateState(CH_ID_TYPE, StringType.valueOf(player.getType()));
+        updateState(CH_ID_CUR_POS, StringType.valueOf("0"));
+        updateState(CH_ID_DURATION, StringType.valueOf("0"));
+        updateState(CH_ID_INPUTS, StringType.valueOf("NULL"));
+        updateState(CH_ID_REPEAT_MODE, StringType.valueOf(player.getRepeatMode()));
+    }
+
+    public class InitializationRunnable implements Runnable {
```

Comment:
This isn't worth to be defined as a public class. Simply put the code in the run method into a lambda and pass that to the executor.